### PR TITLE
KOGITO-1229 list redesign

### PR DESCRIPTION
**Pull request by AjayJagan**
- added icons to states
 - added external link icon
 - added link to process details page in process name
 - added popover for error instances

### CODE REVIEW

**Review by cristianonicolai — APPROVED** [review]

**Review by Sara4994 — APPROVED** [review]

File: packages/management-console/server/MockData/types.js [hunk 1]
```diff
@@ -88,12 +88,12 @@ module.exports = typeDefs = gql`
   }
 
   enum ProcessInstanceState {
-    PENDING
```
Comment (cristianonicolai) [inline packages/management-console/server/MockData/types.js]: The actual enum is upper case on Data Index, changing here will probably break the query, see https://github.com/kiegroup/kogito-apps/blob/master/data-index/data-index-service/src/main/resources/schema.graphqls#L73

Comment (srambach) [reply packages/management-console/server/MockData/types.js]: I have CSS Magic... we can still display it as "Pending" even if it comes in as "PENDING" by using `text-transform: capitalize;` on that element when we put it into the list. 🧙‍♀️

File: packages/management-console/src/components/Molecules/DataToolbarComponent/Datatoolbar.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  text-transform: lowercase;
+}
+
+.pf-c-check__label::first-letter {
+  text-transform: uppercase;
+}
+
+.pf-c-check__label {
+  text-transform: lowercase;
+}
```
Comment (AjayJagan) [inline packages/management-console/src/components/Molecules/DataToolbarComponent/Datatoolbar.css:14]: Hey @srambach - I tried adding text-transform:capitalize , but it was still showing every letter in caps. So I did this work around. I am still trying to figure it out, if I am not successful ,please help me out :)

**Comment by srambach** [comment]
@AjayJagan it's coming into a better shape :) @cristianonicolai 
- All status icons should be default (black) color except for completed (green) and error (red) as those are.
- Spacing of the Apply Filter button is off (I'll comment on the line to fix)
- The name linked to details beside the external icon linked to the endpoint application needs to be addressed. My comments are in the PR I made to your branch. Hopefully there is a name of the endpoint that we could use as text? If not, another option would be to move that to the kebab. But we need to know a little more about how it's used to say for sure - but the details link next to just the external link icon going to the endpoint cannot stay. :-)  https://github.com/AjayJagan/kogito-apps/pull/2

**Comment by srambach** [comment]
OK, I guess it won't let me comment on the line - in DataToolbarComponent.tsx, you can remove the `variant="filter-group"` on line 134:
```
 <DataToolbarGroup variant="filter-group">
```
That's intended to remove the space between multiple filter dropdowns, but the button should be spaced normally.

**Comment by AjayJagan** [comment]
> @AjayJagan it's coming into a better shape :) @cristianonicolai
> 
> * All status icons should be default (black) color except for completed (green) and error (red) as those are.
> * Spacing of the Apply Filter button is off (I'll comment on the line to fix)
> * The name linked to details beside the external icon linked to the endpoint application needs to be addressed. My comments are in the PR I made to your branch. Hopefully there is a name of the endpoint that we could use as text? If not, another option would be to move that to the kebab. But we need to know a little more about how it's used to say for sure - but the details link next to just the external link icon going to the endpoint cannot stay. :-)  [AjayJagan#2](https://github.com/AjayJagan/kogito-apps/pull/2)
I have removed the variant = filter groups and changed the color's of the status.
The only thing that's pending is the endpoint. @cristianonicolai  can you please tell us what should be done regarding that
image

File: packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx [hunk 3]
```diff
@@ -389,136 +390,204 @@ const DataListItemComponent: React.FC<IOwnProps> = ({
...
+                </Link>
+                <a
+                  href={`${processInstanceData.endpoint}/management/processes/${processInstanceData.processId}/instances/${processInstanceData.id}`}
+                  target="_blank"
+                >
+                  <Button variant="link" className="pf-u-pt-0">
+                    <ExternalLinkAltIcon />
+                  </Button>
+                </a>
+              </DataListCell>,
```
Comment (srambach) [inline packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: How about using this to solve the problem of the external icon by itself? And then we also talked about only putting it on the parent, can you do that?
```
              <DataListCell key={1}>
                <Link to={'/ProcessInstances/' + processInstanceData.id} className="pf-u-mr-sm">
                    <strong>{processInstanceData.processName} </strong>
                </Link>
                <Link to={`${processInstanceData.endpoint}/management/processes/${processInstanceData.processId}/instances/${processInstanceData.id}`} target="_blank"  aria-label={`Endpoint for ${processInstanceData.processName}`}>
                  Endpoint <ExternalLinkAltIcon />
                </Link>
              </DataListCell>,
```
@AjayJagan @cristianonicolai

Comment (AjayJagan) [reply packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx]: yep, displaying it only on the parent is possible. awaiting @cristianonicolai response.

**Review by srambach — APPROVED** [review]
I like the endpoint link. Can you make it go to a new tab? @AjayJagan  @cristianonicolai 
Otherwise, approved! Thanks.

**Comment by AjayJagan** [comment]
> I like the endpoint link. Can you make it go to a new tab? @AjayJagan @cristianonicolai
> Otherwise, approved! Thanks.

Done :+1:

**Comment by cristianonicolai** [comment]
@AjayJagan noticed a few things:

- Can you add a mock process with status complete so we can check the UI?
- @srambach  Should the popover close when we click elsewhere? Otherwise we endup with an overlay like in the screenshot below.
- Should buttons in the popover be aligned to the right side? Do we need a close button there?
Selection_032
- In the process details page, error state is not showing up
Selection_033

File: packages/management-console/server/server.js [hunk 4]
```diff
@@ -50,7 +50,7 @@ app.post(
   controller.callRetrigger
 );
 app.post(
-  '/management/processes/:processId/instances/:processInstanceId',
```
Comment (cristianonicolai) [inline packages/management-console/server/server.js]: the endpoint was correct, method should be DELETE.

File: packages/management-console/src/components/Organisms/ProcessDetails/ProcessDetails.tsx [hunk 5]
```diff
@@ -12,14 +12,68 @@ import {
...
 import { Link } from 'react-router-dom';
 
 interface IOwnProps {
   loading: boolean;
   data: any;
 }
 const ProcessDetails: React.FC<IOwnProps> = ({ data, loading }) => {
+  const stateIconCreator = state => {
+    switch (state) {
+      case 'ACTIVE':
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Organisms/ProcessDetails/ProcessDetails.tsx]: would it help to use the TypseScript enum here?

**Review by cristianonicolai — CHANGES_REQUESTED** [review]
see previous comments

File: packages/management-console/server/server.js (hunk 4 above)
Comment (AjayJagan) [reply packages/management-console/server/server.js]: This change is made in KOGITO 789

Comment (cristianonicolai) [reply packages/management-console/server/server.js]: perhaps just revert the change here then.

Comment (AjayJagan) [reply packages/management-console/server/server.js]: done :)

Comment (cristianonicolai) [reply packages/management-console/server/server.js]: sorry but still showing _/abort_, I guess the idea was to just revert the change on this line.

**Review by cristianonicolai — COMMENTED** [review]
@AjayJagan we also need to show the error popup on the process details page, when status is in error.

**Comment by AjayJagan** [comment]
> @AjayJagan we also need to show the error popup on the process details page, when status is in error.

@cristianonicolai  we were adding a view error along with the process management in process details page, So I thought we had to add the pop up instead of the view error. (because the pop contains Skip and Retry which are related to process management)

**Comment by cristianonicolai** [comment]
@AjayJagan we can review the details page in another commit, just revert the change in the abort endpoint and we should be good ti go with this PR.

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**